pkg/controller/controller_binding_test.go

```diff
@@ -1813,8 +1813,8 @@ func TestReconcileUnbindingWithClusterServiceBrokerHTTPError(t *testing.T) {
 	if err := scmeta.AddFinalizer(binding, v1beta1.FinalizerServiceCatalog); err != nil {
 		t.Fatalf("Finalizer error: %v", err)
 	}
-	if err := testController.reconcileServiceBinding(binding); err != nil {
-		t.Fatalf("reconcileServiceBinding should not have returned an error: %v", err)
+	if err := testController.reconcileServiceBinding(binding); err == nil {
```

Comment:
I'm a little surprised that this doesn't affect any of the integration tests, hrm.

Reply:
No current integration tests seem to test that a binding failing at the broker set the resource as Failed. @n3wscott , you were working on binding integration tests; are there more in the pipeline now that were intended to test this behavior?

If not, I can add an integration test that makes sure the resource is not marked as Failed.

Reply:
@pmorie I added an integration test that make sure an unbind that fails is retried and eventually succeeds.